nixos/lib/utils.nix

```diff
@@ -21,8 +21,8 @@ rec {
   toShellPath = shell:
     if types.shellPackage.check shell then
       "/run/current-system/sw${shell.shellPath}"
-    else if types.package.check shell then
-      throw "${shell} is not a shell package"
+    else if isString shell && types.path.check shell then
```

Comment:
Why do we need `isString` when we have `types.path.check` here?  Also, the type of the shell option is `either shellPackage path`, so we know that both test are correct.